src/sema/statements.rs

```diff
@@ -1174,6 +1174,35 @@ fn revert_named_arg(
     }
 }
 
+fn process_event(
+    matches: &bool,
+    ns: &mut Namespace,
+    function_no: &usize,
+    event_no: &usize,
+    event_loc: &pt::Loc,
+    loc: &pt::Loc,
+    cast_args: &[Expression],
+    event_nos: &Vec<usize>,
+    diagnostics: &mut Diagnostics,
+    temp_diagnostics: &Diagnostics,
+) -> Result<Option<Statement>, ()> {
```

Comment:
I think this function could be a closure.


```
let to_stmt =
        |ns: &mut Namespace, event_no: usize, event_loc: pt::Loc, cast_args: Vec<Expression>| {
            if !ns.functions[function_no].emits_events.contains(&event_no) {
                ns.functions[function_no].emits_events.push(event_no);
            }

            Statement::Emit {
                loc: *loc,
                event_no,
                event_loc,
                args: cast_args,
            }
        };
```

Then you can do:
```
               if matches {
                    return Ok(to_stmt(ns, *event_no, event_loc.clone(), cast_args));
                }
                if event_nos.len() > 1 && diagnostics.extend_non_casting(&errors) {
                    return Err(());
                }
```

Reply:
Thanks for the review. I called this closure to avoid duplication.